orca-keel/src/test/java/com/netflix/spinnaker/orca/keel/task/ImportDeliveryConfigTaskTest.java

```diff
@@ -0,0 +1,385 @@
+  public void testTaskResultWhenHttp5xxErrorIsThrown() {
+
+    var mockResponseBody =
+        Map.of(
+            "name",
+            "keeldemo-manifest",
+            "application",
+            "keeldemo",
+            "artifacts",
+            Collections.emptySet(),
+            "environments",
+            Collections.emptySet());
+
+    contextMap.put("attempt", (Integer) contextMap.get("attempt") + 1);
+    contextMap.put(
+        "errorFromLastAttempt",
+        "Retryable HTTP response 500 received from downstream service: HTTP 500 "
+            + wireMock.baseUrl()
+            + "/delivery-configs/: Status: 500, URL: "
+            + wireMock.baseUrl()
+            + "/delivery-configs/, Message: Server Error");
+
+    TaskResult running = TaskResult.builder(ExecutionStatus.RUNNING).context(contextMap).build();
+
+    // Simulate any 5xx http error with empty error response body
+    String emptyBody = "";
+    simulateFault("/delivery-configs/", emptyBody, HttpStatus.INTERNAL_SERVER_ERROR);
+
+    when(scmService.getDeliveryConfigManifest(
+            (String) contextMap.get("repoType"),
+            (String) contextMap.get("projectKey"),
+            (String) contextMap.get("repositorySlug"),
+            (String) contextMap.get("directory"),
+            (String) contextMap.get("manifest"),
+            (String) contextMap.get("ref")))
+        .thenReturn(mockResponseBody);
+
+    var result = importDeliveryConfigTask.execute(stage);
+
+    assertThat(result).isEqualTo(running);
+  }
+
+  @Test
+  public void testTaskResultWhenAPIFailsWithNetworkError() {
+
+    var mockResponseBody =
+        Map.of(
+            "name",
+            "keeldemo-manifest",
+            "application",
+            "keeldemo",
+            "artifacts",
+            Collections.emptySet(),
+            "environments",
+            Collections.emptySet());
+
+    contextMap.put("attempt", (Integer) contextMap.get("attempt") + 1);
+    contextMap.put(
+        "errorFromLastAttempt",
+        String.format(
+            "Network error talking to downstream service, attempt 1 of %s: Connection reset: Connection reset",
+            contextMap.get("maxRetries")));
+
+    TaskResult running = TaskResult.builder(ExecutionStatus.RUNNING).context(contextMap).build();
+
+    // Simulate network failure
+    simulateFault("/delivery-configs/", Fault.CONNECTION_RESET_BY_PEER);
+
+    when(scmService.getDeliveryConfigManifest(
+            (String) contextMap.get("repoType"),
+            (String) contextMap.get("projectKey"),
+            (String) contextMap.get("repositorySlug"),
+            (String) contextMap.get("directory"),
+            (String) contextMap.get("manifest"),
+            (String) contextMap.get("ref")))
+        .thenReturn(mockResponseBody);
+
+    var result = importDeliveryConfigTask.execute(stage);
+
+    assertThat(result).isEqualTo(running);
+  }
+
+  private ImportDeliveryConfigTask.SpringHttpError mockSpringHttpError(
```

Comment:
There's no mocking here, so let's rename the method.
```suggestion
  private ImportDeliveryConfigTask.SpringHttpError makeSpringHttpError(
```

Reply:
The method name is modified to makeSpringHttpError